cupy/cuda/compiler.py

```diff
@@ -758,13 +754,14 @@ def _compile_with_cache_hip(source, options, arch, cache_dir, extra_source,
 
     if cache_dir is None:
         cache_dir = get_cache_dir()
-    # TODO(leofang): it seems as of ROCm 3.5.0 hiprtc/hipcc can automatically
-    # pick up the right arch without needing HCC_AMDGPU_TARGET. Perhaps we just
-    # don't bother passing arch to hiprtc/hipcc?
+    # As of ROCm 3.5.0 hiprtc/hipcc can automatically pick up the
+    # right arch without setting HCC_AMDGPU_TARGET, so we don't need
+    # to tell the compiler which arch we are targeting. But, we still
+    # need to know arch as part of the cache key:
     if arch is None:
-        arch = os.environ.get('HCC_AMDGPU_TARGET')
-        if arch is None:
-            raise RuntimeError('HCC_AMDGPU_TARGET is not set')
+        # On HIP, gcnArch is computed from "compute capability":
+        # https://github.com/ROCm-Developer-Tools/HIP/blob/2080cc113a2d767352b512b9d24c0620b6dee790/rocclr/hip_device.cpp#L202
```

Comment:
```suggestion
        # https://github.com/ROCm-Developer-Tools/HIP/blob/rocm-4.0.0/rocclr/hip_device.cpp#L202
```

Reply:
Oh great, this is better, thanks @kmaehashi!